### PR TITLE
[WIP] Add tests for bugs in `extractCritical`

### PR DESCRIPTION
**Pull request by mattfysh**
**What**: Adding tests for bugs found

**Why**: Bug squashing

**How**: Without fixes (sorry - not enough time!)

**Checklist**:

- [ ] Documentation
- [x] Tests
- [ ] Code complete

### CODE REVIEW

File: test/server.test.js [hunk 1]
```diff
@@ -32,6 +33,9 @@ injectGlobal`
     display: flex;
     justify-content: center;
   }
+  .should-be-included {
+    content: 'css-foo matches regex and this rule is swallowed (current bug)'
+  }
 `
```
Comment (mitchellhamilton) [inline test/server.test.js]: This isn't a bug, this rule isn't included since it'll expand to have the hashed class name which isn't used in the html

Comment (mattfysh) [inline test/server.test.js]: I'm sure it is, right? Change the `css-foo` to be `cs-foo` and it will now appear in the output

Comment (tkh44) [inline test/server.test.js]: the css call returns a class name. Any nested selectors are prefixed with that class name for specificity.

Lets just pretend that that `css`...`` call returns `css-12345`. 

`.should-be-include { ... }` becomes `.css-12345 .should-be-included { ... }`

Comment (mattfysh) [inline test/server.test.js]: This is a call to `injectGlobal`... not `css`. If you're viewing on GitHub, you might be having a few lines between getting clipped and that's why you're seeing it as a call to `css`

But ignore that for now - please just watch the bug in motion by changing `css-foo` in the content value to `cs-foo` and you'll now see this rule coming through in `extractCritical`.

Comment (tkh44) [inline test/server.test.js]: Oh man. My bad on that. It looks like you are correct. @mitchellhamilton you mind taking one more look at this?

Comment (mitchellhamilton) [inline test/server.test.js]: I'm so sorry, you're right @mattfysh

Comment (mattfysh) [inline test/server.test.js]: No worries, just a minor edge case. I guess you could leave the test in pending (`xtest()`?) and fix it if/when you want to support
